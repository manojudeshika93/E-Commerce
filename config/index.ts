export * from './locale.config';
export * from './tw';

