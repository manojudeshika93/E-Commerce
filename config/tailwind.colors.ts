export default {
  primary: {
    500: "#C30145",
    400: "#BB4242",
    300: "#AD4C4C",
    200: "#9E5858",
    100: "#724040",
  },
  secondary: {
    600: "#404040",
    500: "#707070",
    400: "#A0A0A0",
    300: "#CFCFCF",
    200: "#E7E7E7",
    100: "#F3F3F3",
  },
  status: {
    success: {
      DEFAULT: "#44B388",
    },
    warning: {
      DEFAULT: "#EAB125",
    },
    error: {
      DEFAULT: "#D63535",
    },
    info: {
      DEFAULT: "#45A2FF",
    },
  },
  custom: {
    transparent: "#00000000",
    white: "#FFFFFF",
    black: "#000000",
  },
};
