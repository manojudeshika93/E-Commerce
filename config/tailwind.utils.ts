const textLevels = [
  { level: 'h1', fontSize: 40, lineHeight: 48 },
  { level: 'h2', fontSize: 30, lineHeight: 40 },
  { level: 'h3', fontSize: 20, lineHeight: 30 },
  { level: 'b1', fontSize: 16, lineHeight: 24 },
  { level: 'b2', fontSize: 14, lineHeight: 22 },
  { level: 'b3', fontSize: 12, lineHeight: 20 },
];

const fontFamilies = {
  regular: 'Inter_400Regular',
  medium: 'Inter_500Medium',
  semibold: 'Inter_600SemiBold',
  bold: 'Inter_700Bold',
};

const styles: Record<string, any> = {};

textLevels.forEach(({ level, fontSize, lineHeight }) => {
  Object.entries(fontFamilies).forEach(([name, fontFamily]) => {
    styles[`.text-${level}-${name}`] = {
      fontSize,
      lineHeight,
      fontFamily,
    };
  });
});

textLevels.forEach(({ level, fontSize, lineHeight }) => {
  styles[`.text-${level}`] = {
    fontSize,
    lineHeight,
    fontFamily: fontFamilies.regular,
  };
});

styles['center-full'] = `justify-center items-center`;

export default styles;
