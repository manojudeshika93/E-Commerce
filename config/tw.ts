import { create } from 'twrnc';

import color from './tailwind.colors';

export const tw = create(require('../tailwind.config'));

export const Color = color;
