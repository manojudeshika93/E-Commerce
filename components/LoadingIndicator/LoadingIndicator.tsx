import React from 'react';
import { ActivityIndicator as RNLoadingIndicator, View } from 'react-native';

import { Color, tw } from '@/config';

export interface LoadingIndicatorProps {
  color?: string;
  size?: 'small' | 'large';
}

export function LoadingIndicator({ color = Color.primary[500], size = 'large' }: LoadingIndicatorProps) {
  return <View style={tw`w-full h-full items-center justify-center`}><RNLoadingIndicator color={color} size={size} animating={true} /></View>;
}
