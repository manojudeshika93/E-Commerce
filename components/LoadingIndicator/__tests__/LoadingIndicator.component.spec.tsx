import { render } from '@testing-library/react-native';
import * as React from 'react';

import { Color } from '@/config';

import { LoadingIndicator, LoadingIndicatorProps } from '../LoadingIndicator';

describe('LoadingIndicator Component', () => {
  const propSet = [
    {
      color: Color.primary[500],
      size: 'small',
    },
    {
      color: Color.secondary[100],
      size: 'large',
    },
    {
      color: Color.custom.black,
      size: 'large',
    },
  ];

  const getRenderedComponent = (props?: Partial<LoadingIndicatorProps>) => render(<LoadingIndicator {...props} />);

  describe('should render correctly', () => {
    for (const prop of propSet) {
      it(`should render correctly with ${prop}`, () => {
        const rendered = getRenderedComponent({ ...prop } as any);
        const renderedTree = rendered.toJSON();
        expect(renderedTree).toMatchSnapshot();
      });
    }
  });
});
