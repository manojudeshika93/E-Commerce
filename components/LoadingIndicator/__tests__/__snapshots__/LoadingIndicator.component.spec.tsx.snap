// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator Component should render correctly should render correctly with [object Object] 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#C30145"
    size="small"
  />
</View>
`;

exports[`LoadingIndicator Component should render correctly should render correctly with [object Object] 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#F3F3F3"
    size="large"
  />
</View>
`;

exports[`LoadingIndicator Component should render correctly should render correctly with [object Object] 3`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#000000"
    size="large"
  />
</View>
`;
